From 1813371cce9a969334747e08be36d95d3af05cf1 Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Thu, 23 Aug 2018 11:03:29 +0800
Subject: [PATCH 4438/5242] MLK-19305-1 soc: imx: add PAD wakeup support

commit  da291e8f6b52fb52d3f1838c80f4f02a77d9e9ac from
https://source.codeaurora.org/external/imx/linux-imx.git

Add PAD wakeup support for i.MX8 platforms with system
controller present, with PAD wakeup feature enabled,
the corresponding resource's power is no need to be
kept enabled when linux suspend, thus save a sub-system's
power consumption.

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Reviewed-by: Bai Ping <ping.bai@nxp.com>
(cherry picked from commit 3b5d781273b22461de9aaea337f9da9b2fdb643e)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/soc/imx/sc/main/ipc.c     |    5 +++++
 include/soc/imx8/sc/svc/irq/api.h |    1 +
 2 files changed, 6 insertions(+)

diff --git a/drivers/soc/imx/sc/main/ipc.c b/drivers/soc/imx/sc/main/ipc.c
index cfe70ec..65af00b 100644
--- a/drivers/soc/imx/sc/main/ipc.c
+++ b/drivers/soc/imx/sc/main/ipc.c
@@ -408,6 +408,11 @@ int __init imx8_mu_init(void)
 	if (sciErr)
 		pr_info("Cannot request WDOG interrupt\n");
 
+	sciErr = sc_irq_enable(mu_ipcHandle, SC_R_MU_1A, SC_IRQ_GROUP_WAKE,
+			       SC_IRQ_PAD, true);
+	if (sciErr)
+		pr_info("Cannot request PAD interrupt\n");
+
 	register_syscore_ops(&imx8_mu_syscore_ops);
 
 	pr_info("*****Initialized MU\n");
diff --git a/include/soc/imx8/sc/svc/irq/api.h b/include/soc/imx8/sc/svc/irq/api.h
index bd1b7fc..e416ffa 100644
--- a/include/soc/imx8/sc/svc/irq/api.h
+++ b/include/soc/imx8/sc/svc/irq/api.h
@@ -70,6 +70,7 @@
  */
 /*@{*/
 #define SC_IRQ_WDOG              (1 << 0)	/* Watchdog interrupt */
+#define SC_IRQ_PAD               (1U << 1U)     /*!< Pad wakeup */
 /*@}*/
 
 /*!
-- 
1.7.9.5

